src/core/Akka.Streams/Implementation/Fusing/GraphInterpreter.cs

```diff
@@ -344,7 +344,8 @@ public sealed class Connection
         // Using an Object-array avoids holding on to the GraphInterpreter class
         // when this accidentally leaks onto threads that are not stopped when this
         // class should be unloaded.
-        private static readonly ThreadLocal<object[]> CurrentInterpreter = new ThreadLocal<object[]>(() => new object[1]);
+        [ThreadStatic]
+        private static object[] CurrentInterpreter;
```

Comment:
Changing to ThreadStatic because of https://ayende.com/blog/189761-A/production-postmortem-the-slow-slowdown-of-large-systems